### DISCUSSION TITLE
cant build capacitor app, provisioning profile error

### DISCUSSION THREAD

**Discussion by luddthree** [codemagic-ci-cd/codemagic-docs#2476] [Q&A]
### If you are looking for help, please confirm the following...

- [x] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Ionic/Cordova

### Steps to reproduce

i built my app in nuxt3 and used capacitor js. 
repo: https://github.com/luddthree/vorspill

i have all the provisioning profiles on both the apple developer portal and in codemagic, all of the identifyers add up, i dont understand.
I use arch btw and dont have a mac (if that is any helpfull to know). 

### Expected results

build the app 

### Actual results

❌  /Users/builder/clone/ios/App/App.xcodeproj: error: "App" requires a provisioning profile. Select a provisioning profile in the Signing & Capabilities editor. (in target 'App' from project 'App')

    Run script build phase '[CP] Embed Pods Frameworks' will be run during every build because it does not specify any outputs. To address this warning, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'App' from project 'App')
Failed to archive App.xcworkspace

### Build id (optional)

661d329785bdf9a88c292572

**Comment by dtrdic** [comment]
Hi @luddthree open your iOS project in an editor and check the `project.pbxproj` file. 
In there set the `PRODUCT_BUNDLE_IDENTIFIER` to the bundle id you had just created. This is mentioned in multiple places in the project, so make sure to change it everywhere. 

Also add the archive flag to your build ipa command;
```
      - name: Build ipa for distribution
        script: |
          xcode-project build-ipa \
            --workspace "$XCODE_WORKSPACE" \
            --scheme "$XCODE_SCHEME" \
            --archive-flags="-destination 'generic/platform=iOS'" 
```

**Comment by dtrdic** [comment]
I'm tentatively closing the discussion since we haven't heard back from you. You can open the discussion again by replying to this message or opening a new discussion.
